promptsource/seqio_tasks/tasks.py

```diff
@@ -42,26 +41,12 @@ def strip_whitespace(output_or_target, example=None, is_target=False):
     return output_or_target.strip()
 
 
-def get_label_strings(template):
-    target = template.jinja.split("|||")[1]
-    label_list_re = r"^([^\{\}]*)\{\{\s*(\[\s*[\"|\'].*[\"|\']\s*\])\s*\[.*\]\s*\}\}([^\{\}]*)$"
-    label_string_match = re.search(label_list_re, target.strip())
-
-    if label_string_match:
-        before_label = label_string_match.group(1)
-        labels = eval(label_string_match.group(2))
-        after_label = label_string_match.group(3)
-        labels = [before_label + label + after_label for label in labels]
-        return labels
-
-
 def maybe_get_class_id_postprocessor(template):
-    labels = get_label_strings(template)
-    if labels is not None:
+    if template.answer_choices:
```

Comment:
The intended usage was that the answer choices would always be obtained through `get_answer_choices_list(example)`, but this function doesn't have access to a particular example. The issue with this is that we planned to merge answer_choices and answer_choices_key, since they can be represented in the same syntax (a jinja expression). So this access to a list without a specific example was intended to go away before release. Is that a problem?

Reply:
So, the problem is that there really are kind of two cases for answer choices:
1. They are static across the entire dataset
2. They vary on a per-example basis (and possibly even vary in length)

(1) is actually I think a lot more common than (2). We need to make the assumption that answer_choices is fixed-length to do things like "map label string to an integer" (e.g. "Yes" -> 0, "No" -> 1), which is necessary for certain metrics. If we force everything into the general case (2) we won't be able to do this. And for case (2), we don't care about doing the things that we want to do with (1), since we can't e.g. map label strings to integers if the label strings differ per-example. So, if we do get rid of answer_choices, it'd be important to still allow things for (1). Not sure the best way to support that.

Reply:
Ah darn, sorry, it seemed like a good idea at the time to combine them all into one representation on the annotation side. What if I added a function to templates like `get_fixed_answer_choices_list()` that either returns the fixed answer choices or None if it's not fixed? I would just see if the expression parses without access to any example, otherwise return None.

Reply:
That would work great I think.

Reply:
Updated code base on #423